ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,152 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Upload commands.
+"""
+
+import os
+import uuid
+from contextlib import closing
+from argparse import ArgumentTypeError
+
+from .. _internal import qemu_img
+from .. _internal.units import KiB, MiB
+
+from . import _api
+from . import _options
+from . import _ovirt
+from . import _ui
+
+
+def register(parser):
+    cmd = parser.add_sub_command(
+        "upload-disk",
+        help="Upload disk",
+        func=upload_disk)
+
+    cmd.add_argument(
+        "--sd-name",
+        required=True,
+        help="Name of the storage domain.")
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Upload image format (default qcow2).")
+
+    cmd.add_argument(
+        "--sparse",
+        action="store_true",
+        help="Create sparse disk. Cannot be used with raw format on "
+             "block storage.")
```

Comment:
Do we want to validate this? Or other things like the initial_size being big enough? Or do we just leave that to the server?

Reply:
The server will reject invalid combinations, same with backup and format.